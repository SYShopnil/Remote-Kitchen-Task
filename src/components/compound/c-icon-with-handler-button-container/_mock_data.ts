import {
  EMode,
  ICIconWithHandlerButtonContainer,
} from "@src/types/compound/c-icon-with-handler-button-container";

export const mockDataForTest: ICIconWithHandlerButtonContainer[] = [
  {
    foodId: "T_e_x_a_s_ _Z_i_n_g_e_r_ _B_e_l_l_17814346472074",
    mode: EMode.UPDATE,
    existFoodData: {
      name: "Texas Zinger Bell",
      desc: "Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia,\nmolestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum\nnumquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium\noptio, eaque rerum! Provident similique accusantium nemo autem. Veritatis\nobcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam\nnihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit,\ntenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit,",
      image:
        "/assert/T_e_x_a_s_ _Z_i_n_g_e_r_ _B_e_l_l_17143464720741714346472074.jpg",
      price: "25",
      foodId: "T_e_x_a_s_ _Z_i_n_g_e_r_ _B_e_l_l_17814346472074",
    },
  },
  {
    foodId: "T_e_x_a_s_ _Z_i_n_g_e_r_ _B_e_l_l_17814346472074",
    mode: EMode.DELETE,
  },
];
