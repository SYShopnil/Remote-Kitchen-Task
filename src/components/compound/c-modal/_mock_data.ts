import { ICModal } from "@src/types/compound/c-modal";
import React from "react";

export const mockDataForCModal: ICModal = {
  isOpen: true,
  setIsOpen: () => {
    return true;
  },
  title: "For Test",
  children: React.createElement("h1", "Hello"),
};
