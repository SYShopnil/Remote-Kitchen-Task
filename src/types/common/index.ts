export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EAuth {
  AuthTokenCookieName = "auth",
}

export enum EDataTestId {
  SHeaderMain = "SHeaderMain",
  SProductSection = "SProductSection",
  SProfile = "SProfile",
  CPaginationTrack = "CPaginationTrack",
  RSingleProduct = "RSingleProduct",
  RTableBody = "RTableBody",
  SProductCard = "SProductCard",
  CIconWithHandlerButtonContainer = "CIconWithHandlerButtonContainer",
  CModal = "CModal",
  CProductAddSection = "CProductAddSection",
  CProductSearchBarContainer = "CProductSearchBarContainer",
  CForm = "CForm",
}
