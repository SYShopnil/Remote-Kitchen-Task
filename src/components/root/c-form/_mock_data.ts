import { BtnColorSchema } from "@src/types/root";
import { FormElementEnum, IForm } from "@src/types/root/c-form";

export const mockProps: IForm = {
  buttonInfo: {
    btnText: "Demo One",
    colorSchema: BtnColorSchema.SolidBgGrayTextViolet,
    isArrow: false,
  },
  formFields: [
    {
      elementType: FormElementEnum.Input,
      label: "Food Name",
      space: 2,
      isRequired: true,
      name: "name",
      type: "input",
      placeholder: "Give Name",
    },
    {
      elementType: FormElementEnum.Input,
      label: "Food Price",
      space: 2,
      isRequired: true,
      name: "price",
      type: "input",
      placeholder: "Give Price",
    },
    {
      elementType: FormElementEnum.TextArea,
      label: "Description",
      space: 2,
      isRequired: true,
      name: "desc",
      type: "input",
      placeholder: "Give Description",
    },
    {
      elementType: FormElementEnum.Upload,
      label: "Upload Food Picture",
      space: 2,
      isRequired: true,
      name: "image",
      type: "input",
      placeholder: "Give image",
    },
  ],
  onSubmit: () => {},
  formTitle: "Test Form",
};
