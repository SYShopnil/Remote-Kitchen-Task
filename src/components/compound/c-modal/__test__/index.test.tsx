import { render, screen } from "@testing-library/react";
import { EDataTestId } from "@src/types/common";
import { CModal } from "..";
import { mockDataForCModal } from "../_mock_data";

describe("When Modal has rendered", () => {
  it("Expect it renders successfully", () => {
    const { getByTestId } = render(
      <CModal {...mockDataForCModal}>
        <h1>Hello world</h1>
      </CModal>
    );
    const myElement = getByTestId(EDataTestId.CModal);
    // const myElement = screen.getByTestId(EDataTestId.CIconWithHandlerButtonContainer);
    expect(myElement).toMatchSnapshot();
  });
});
