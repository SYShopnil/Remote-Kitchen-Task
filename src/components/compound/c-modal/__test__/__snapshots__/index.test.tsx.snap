// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Modal has rendered Expect it renders successfully 1`] = `
<div
  class="fixed z-10 inset-0 overflow-y-auto"
  data-testid="CModal"
>
  <div
    class="flex items-center justify-center min-h-screen p-4"
  >
    <div
      aria-hidden="true"
      class="fixed inset-0 transition-opacity"
    >
      <div
        class="absolute inset-0 bg-black opacity-50"
      />
    </div>
    <div
      aria-labelledby="modal-title"
      aria-modal="true"
      class="bg-white rounded-lg overflow-hidden transform transition-all max-w-lg w-full"
      role="dialog"
    >
      <div
        class="p-4"
      >
        <h2
          class="text-xl font-bold mb-4 "
          id="modal-title"
        >
          For Test
        </h2>
        <div
          class="text-left"
        >
          <h1>
            Hello world
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
